src/ui/svgWidget.js

```diff
@@ -0,0 +1,78 @@
+//////////////////////////////////////////////////////////////////////////////
+/**
+ * Create a new instance of class geo.gui.svgWidget
+ *
+ * Due to the nature of d3 creating DOM elements as it inserts them, calls to appendChild
+ * don't appear in this widget.
+ *
+ * The canvas of an svgWidget always refers to the actual svg element.
+ * The parentCanvas can refer to another widgets svg element, dom element, or the
+ * UI layers dom element.
+ * See {@link geo.gui.widget#parentCanvas}.
+ *
+ * @class
+ * @extends geo.gui.domWidget
+ * @returns {geo.gui.svgWidget}
+ *
+ */
+//////////////////////////////////////////////////////////////////////////////
+geo.gui.svgWidget = function (arg) {
+  'use strict';
+  if (!(this instanceof geo.gui.svgWidget)) {
+    return new geo.gui.svgWidget(arg);
+  }
+
+  geo.gui.domWidget.call(this, arg);
+
+  var m_this = this,
+      m_renderer = null;
+
+  this._init = function (arg) {
+    var d3Parent;
+    arg = m_this.parseArgs(arg);
+    m_this.args = arg;
```

Comment:
This inherits from `widget` though.  `svgWidget` and `domWidget` are sibling classes.  You could even move the options parsing out of `_init` and put it directly in the constructor.  This is what is done in [`geo.map`](https://github.com/OpenGeoscience/geojs/blob/master/src/core/map.js#L68).  The subclasses all get the same `arg` object anyway so as long as you mutate it, the subclasses will automatically get the changes.

If I'm missing something, I don't have a huge problem with how this is done now, but I would want to save `args` as `m_this._args` instead to imply that it is a protected value.